website/docs/extensions/write/onboarding-workflow.md

```diff
@@ -0,0 +1,531 @@
+
+Adding onboarding to an extension is as simple as writing JSON in the `package.json`. Podman Desktop will convert the JSON object into actual code to render all items.
+
+Onboarding consists of a title, a description, media (image), an enablement clause, and a list of steps. Before getting into the details, let's examine the JSON schema.
+
+```json
+{
+  "title": "onboarding",
+  "type": "object",
+  "properties": {
+    "title": {
+      "type": "string"
+    },
+    "description": {
+      "type": "string"
+    },
+    "media": {
+      "path": {
+        "type": "string"
+      },
+      "altText": {
+        "type": "string"
+      }
+    },
+    "enablement": {
+      "type": "string"
+    },
+    "steps": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "type": "string"
+          },
+          "title": {
+            "type": "string"
+          },
+          "description": {
+            "type": "string"
+          },
+          "media": {
+            "path": {
+              "type": "string"
+            },
+            "altText": {
+              "type": "string"
+            }
+          },
+          "command": {
+            "type": "string"
+          },
+          "completionEvents": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+          },
+          "content": {
+            "type": "array",
+            "items": {
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "value": {
+                    "type": "string"
+                  },
+                  "highlight": {
+                    "type": "boolean"
+                  },
+                  "when": {
+                    "type": "string"
+                  }
+                },
+                "required": ["value"]
+              }
+            }
+          },
+          "when": {
+            "type": "string"
+          },
+          "component": {
+            "type": "string",
+            "enum": ["createContainerProviderConnection", "createKubernetesProviderConnection"]
+          },
+          "state": {
+            "type": "string",
+            "enum": ["completed", "failed"]
+          }
+        },
+        "required": ["id", "title"]
+      }
+    }
+  },
+  "required": ["title", "enablement", "steps"]
+}
+```
+
+### Title, Description and Media
+
+The **title**, the **description** and the **media** are all placed in the top left of the onboarding page.
+Only the title is required. The description and the media are optional.
+If the media is not specified, Podman Desktop will display the default extension icon.
```

Comment:
'the default extension icon' - clarify so it's clear whether this means a generic extension icon (e.g. puzzle piece) or the icon of the extension.